### PR TITLE
En-5761 : unit tests auction staking smart contract

### PR DESCRIPTION
**Pull request by miiu96**

### CODE REVIEW

File: vm/systemSmartContracts/auction_test.go [hunk 1]
```diff
@@ -48,6 +50,80 @@ func createABid(totalStakeValue uint64, numBlsKeys uint32, maxStakePerNode uint6
 	return data
 }
 
+func TestStakingAuctionSmartContract_NilValidatorSetting(t *testing.T) {
```
Comment (bogdan-rosianu) [inline vm/systemSmartContracts/auction_test.go]: Test**New**Staking... ? also in the tests below

File: vm/systemSmartContracts/auction_test.go [hunk 2]
```diff
@@ -1112,3 +1188,247 @@ func TestAuctionStakingSC_ExecuteGetShouldOk(t *testing.T) {
...
+	blockChainHook := &mock.BlockChainHookStub{}
+	args := createMockArgumentsForAuction()
+	args.Eei = createVmContextWithStakingSc(minStakeValue, unboundPeriod, blockChainHook)
+
+	sc, _ := NewStakingAuctionSmartContract(args)
+
+	//change reward address should error nil arguments
+	changeRewardAddress(t, sc, stakerAddress, nil, vmcommon.UserError)
+	// change reward address should error wrong address
+	changeRewardAddress(t, sc, stakerAddress, []byte("a"), vmcommon.UserError)
```
Comment (bogdan-rosianu) [inline vm/systemSmartContracts/auction_test.go]: maybe "wrong address" instead of "a" ?

File: vm/systemSmartContracts/auction_test.go (hunk 1 above)
Comment (miiu96) [reply vm/systemSmartContracts/auction_test.go]: done.

File: vm/systemSmartContracts/auction_test.go (hunk 2 above)
Comment (miiu96) [reply vm/systemSmartContracts/auction_test.go]: done.

**Pull request merged**